bmc/boot_device.go

```diff
@@ -78,3 +95,67 @@ func SetBootDeviceFromInterfaces(ctx context.Context, timeout time.Duration, boo
 	}
 	return setBootDevice(ctx, timeout, bootDevice, setPersistent, efiBoot, bdSetters)
 }
+
+// getBootDeviceOverride gets the boot device override settings for the given provider,
+// and updates the given metadata with provider attempts and errors.
+func getBootDeviceOverride(
+	ctx context.Context,
+	timeout time.Duration,
+	provider *bootOverrideProvider,
+	metadata *Metadata,
+) (override *BootDeviceOverride, err error) {
+	select {
+	case <-ctx.Done():
+		err = multierror.Append(err, ctx.Err())
+		return nil, err
+	default:
+		metadata.ProvidersAttempted = append(metadata.ProvidersAttempted, provider.name)
+		ctx, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+
+		override, err = provider.bootOverrider.BootDeviceOverrideGet(ctx)
+		if err != nil {
+			metadata.FailedProviderDetail[provider.name] = err.Error()
+			return nil, nil
+		}
+
+		metadata.SuccessfulProvider = provider.name
+		return override, nil
+	}
+}
+
+// GetBootDeviceOverrideFromInterface will get boot device override settings from the first successful
+// call to a BootDeviceOverrideGetter in the array of providers.
+func GetBootDeviceOverrideFromInterface(
+	ctx context.Context,
+	timeout time.Duration,
+	providers []interface{},
+) (*BootDeviceOverride, Metadata, error) {
+	var err error
+	metadata := Metadata{
```

Comment:
for consistency use the [`newMetadata()`](https://github.com/bmc-toolbox/bmclib/blob/b0ef181d6ed441ea5bd9e7161e92712633a28811/bmc/bmc.go#L24) helper method here

Reply:
Done